Lib/xml/etree/ElementTree.py

```diff
@@ -1200,6 +1200,15 @@ def parse(source, parser=None):
     Return an ElementTree instance.
 
     """
+    import warnings
+
+    if (hasattr(source, 'encoding') and
+        source.encoding == 'ISO-8859-1' and
+        source.mode == 'r'):
+        warnings.warn(
```

Comment:
Please move the "import warnings" here.

Reply:
It seems more perfect this way. I've made changes :)

Reply:
Moreover, why not just `if getattr(source, 'encoding', None) == 'r':`?